packages/trader/src/AppV2/Components/TradeParameters/TakeProfit/take-profit.tsx

```diff
@@ -1,26 +1,216 @@
 import React from 'react';
 import clsx from 'clsx';
 import { observer } from 'mobx-react';
-import { TextField } from '@deriv-com/quill-ui';
+import { ActionSheet, SectionMessage, TextField, Text, ToggleSwitch, TextFieldWithSteppers } from '@deriv-com/quill-ui';
 import { Localize, localize } from '@deriv/translations';
 import { useTraderStore } from 'Stores/useTraderStores';
-import { getCurrencyDisplayCode } from '@deriv/shared';
+import { getCurrencyDisplayCode, getDecimalPlaces } from '@deriv/shared';
+import { focusAndOpenKeyboard } from 'AppV2/Utils/trade-params-utils';
+import Carousel from 'AppV2/Components/Carousel';
+import TakeProfitHeader from './take-profit-header';
 
 type TTakeProfitProps = {
     is_minimized?: boolean;
 };
 
 const TakeProfit = observer(({ is_minimized }: TTakeProfitProps) => {
-    const { currency, has_open_accu_contract, take_profit } = useTraderStore();
+    const {
+        currency,
+        has_open_accu_contract,
+        has_take_profit,
+        is_accumulator,
+        take_profit,
+        onChangeMultiple,
+        onChange,
+        validation_params,
+    } = useTraderStore();
+
+    const [is_open, setIsOpen] = React.useState(false);
+    const [is_take_profit_enabled, setIsTakeProfitEnabled] = React.useState(has_take_profit);
```

Comment:
```suggestion
    const [is_enabled, setIsEnabled] = React.useState(has_take_profit);
```

Reply:
This would make more sense since the component is already about take-profit only, but I see there's another `is_enabled` in this file

Reply:
Agree, will apply